modules/markup/openapi/openapi.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package openapi
+
+import (
+	"fmt"
+	"io"
+	"net/url"
+
+	"code.gitea.io/gitea/modules/markup"
+	"code.gitea.io/gitea/modules/setting"
+
+	"github.com/gobwas/glob"
+)
+
+func init() {
+	markup.RegisterRenderer(Renderer{})
+}
+
+// Renderer implements markup.Renderer for asciicast files.
+// See https://github.com/asciinema/asciinema/blob/develop/doc/asciicast-v2.md
+type Renderer struct{}
+
+var _ markup.GlobMatchRenderer = (*Renderer)(nil)
+
+// Name implements markup.Renderer
+func (Renderer) Name() string {
+	return "openapi"
+}
+
+// SanitizerDisabled disabled sanitize if return true
+func (Renderer) SanitizerDisabled() bool {
+	return true
+}
+
+func (Renderer) DisplayInNewPage() bool {
+	return true
+}
+
+func (Renderer) MatchGlobs() []glob.Glob {
+	return []glob.Glob{
+		glob.MustCompile("**{openapi,OpenAPI,swagger}.{yml,yaml,json,JSON,Yaml,YML}", '/'),
+	}
+}
```

Comment:
It seems that `MatchGlobs` is not a general approach.

The render could provide a function like "CanRenderFileName(filename string) bool", then every render can have their own rules to detect whether a file can be rendered.